src/Symfony/Component/Messenger/Asynchronous/Routing/AbstractSenderLocator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Asynchronous\Routing;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+abstract class AbstractSenderLocator implements SenderLocatorInterface
```

Comment:
What's the reasoning for using an abstract class over a trait for this `static` method?